### PR TITLE
Nojima/hotpot 1703 more strict 2

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

File: shared/actions/settings.tsx [hunk 1]
```diff
@@ -364,7 +366,7 @@ const loadSettings = async (
       (settings.emails ?? []).map(row => [row.email, {...Constants.makeEmailRow(), ...row}])
     )
     const phoneMap = (settings.phoneNumbers ?? []).reduce<Map<string, Types.PhoneRow>>((map, row) => {
-      if (map[row.phoneNumber] && !map[row.phoneNumber].superseded) {
+      if (map.get(row.phoneNumber) && !map.get(row.phoneNumber)?.superseded) {
```
Comment (chrisnojima) [inline shared/actions/settings.tsx]: actual bug!

File: shared/actions/unlock-folders.tsx [hunk 2]
```diff
@@ -49,11 +49,13 @@ const delegateRekeyUI = (action: EngineGen.Keybase1RekeyUIDelegateRekeyUIPayload
     dangling: true,
     incomingCallMap: {
       'keybase.1.rekeyUI.refresh': ({sessionID, problemSetDevices}) =>
-        UnlockFoldersGen.createNewRekeyPopup({
-          devices: problemSetDevices.devices || [],
-          problemSet: problemSetDevices.problemSet,
-          sessionID,
-        }),
+        Saga.put(
```
Comment (chrisnojima) [inline shared/actions/unlock-folders.tsx]: actual bug!

File: shared/actions/teams.tsx [hunk 3]
```diff
@@ -1083,7 +1084,8 @@ function* setPublicity(state: TypedState, action: TeamsGen.SetPublicityPayload)
     )
   }
 
-  const results = yield Saga.all(calls)
+  // TODO fix type
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/actions/teams.tsx]: No just in the future

File: shared/actions/platform-specific/index.native.tsx [hunk 4]
```diff
@@ -29,13 +29,15 @@ import {
 } from 'react-native'
 import CameraRoll from '@react-native-community/cameraroll'
 import NetInfo from '@react-native-community/netinfo'
+// @ts-ignore
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/actions/platform-specific/index.native.tsx]: If it's ever an import it's cause the types don't exist

[the remark this answers is not in the document]
Comment (hyperobject) [reply shared/actions/platform-specific/index.native.tsx]: yeah I figured as much, bummer

**Pull request merged**